### ISSUE TITLE
server: data race in MarshalJson

### ISSUE THREAD

**Issue by RaduBerinde** [cockroachdb/cockroach#69509]
Seeing testrace failures in `security` package with the data race below:

```
WARNING: DATA RACE
Read at 0x00c0005bf6e8 by goroutine 105:
  github.com/cockroachdb/cockroach/pkg/util/metric.(*Registry).MarshalJSON()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/metric/registry.go:161 +0x88
  encoding/json.marshalerEncoder()
      /usr/local/go/src/encoding/json/encode.go:479 +0xc3
  encoding/json.(*encodeState).reflectValue()
      /usr/local/go/src/encoding/json/encode.go:360 +0x93
  encoding/json.interfaceEncoder()
      /usr/local/go/src/encoding/json/encode.go:716 +0xfb
  encoding/json.mapEncoder.encode()
      /usr/local/go/src/encoding/json/encode.go:813 +0x3cc
  encoding/json.mapEncoder.encode-fm()
      /usr/local/go/src/encoding/json/encode.go:779 +0x88
  encoding/json.(*encodeState).reflectValue()
      /usr/local/go/src/encoding/json/encode.go:360 +0x93
  encoding/json.(*encodeState).marshal()
      /usr/local/go/src/encoding/json/encode.go:332 +0xd2
  encoding/json.Marshal()
      /usr/local/go/src/encoding/json/encode.go:161 +0x78
  github.com/cockroachdb/cockroach/pkg/server/status.(*MetricsRecorder).MarshalJSON()
      /go/src/github.com/cockroachdb/cockroach/pkg/server/status/recorder.go:239 +0x450
  encoding/json.marshalerEncoder()
      /usr/local/go/src/encoding/json/encode.go:479 +0xc3
  encoding/json.(*encodeState).reflectValue()
      /usr/local/go/src/encoding/json/encode.go:360 +0x93
  encoding/json.(*encodeState).marshal()
      /usr/local/go/src/encoding/json/encode.go:332 +0xd2
  encoding/json.Marshal()
      /usr/local/go/src/encoding/json/encode.go:161 +0x78
  encoding/json.MarshalIndent()
      /usr/local/go/src/encoding/json/encode.go:176 +0x64
  github.com/cockroachdb/cockroach/pkg/server.marshalToJSON()
      /go/src/github.com/cockroachdb/cockroach/pkg/server/status.go:2573 +0xee
  github.com/cockroachdb/cockroach/pkg/server.marshalJSONResponse()
      /go/src/github.com/cockroachdb/cockroach/pkg/server/status.go:2583 +0x177
  github.com/cockroachdb/cockroach/pkg/server.(*statusServer).Metrics()
      /go/src/github.com/cockroachdb/cockroach/pkg/server/status.go:1509 +0x13c
  github.com/cockroachdb/cockroach/pkg/server/serverpb._Status_Metrics_Handler.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/server/serverpb/status.pb.go:5936 +0xa4
  github.com/cockroachdb/cockroach/pkg/util/tracing.ServerInterceptor.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/tracing/grpc_interceptor.go:126 +0x67c
  google.golang.org/grpc.getChainUnaryHandler.func1()
      /go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1134 +0x14f
  github.com/cockroachdb/cockroach/pkg/rpc.NewServer.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/rpc/context.go:177 +0x124
  google.golang.org/grpc.getChainUnaryHandler.func1()
      /go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1134 +0x14f
  github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.unaryInterceptor()
      /go/src/github.com/cockroachdb/cockroach/pkg/rpc/auth.go:71 +0x138
  github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.unaryInterceptor-fm()
      /go/src/github.com/cockroachdb/cockroach/pkg/rpc/auth.go:47 +0xa4
  google.golang.org/grpc.chainUnaryServerInterceptors.func1()
      /go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1120 +0x112
  github.com/cockroachdb/cockroach/pkg/server/serverpb._Status_Metrics_Handler()
      /go/src/github.com/cockroachdb/cockroach/pkg/server/serverpb/status.pb.go:5938 +0x1f1
  google.golang.org/grpc.(*Server).processUnaryRPC()
      /go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1293 +0xa65
  google.golang.org/grpc.(*Server).handleStream()
      /go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1618 +0x142b
  google.golang.org/grpc.(*Server).serveStreams.func1.2()
      /go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:941 +0xe6

Previous write at 0x00c0005bf6e8 by goroutine 126:
  github.com/cockroachdb/cockroach/pkg/util/metric.(*Registry).AddMetric()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/metric/registry.go:72 +0x146
  github.com/cockroachdb/cockroach/pkg/util/metric.(*Registry).addMetricValue()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/metric/registry.go:129 +0x385
  github.com/cockroachdb/cockroach/pkg/util/metric.(*Registry).AddMetricStruct()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/metric/registry.go:109 +0x49c
  github.com/cockroachdb/cockroach/pkg/util/metric.(*Registry).addMetricValue()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/metric/registry.go:131 +0x3e5
  github.com/cockroachdb/cockroach/pkg/util/metric.(*Registry).AddMetricStruct()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/metric/registry.go:109 +0x49c
  github.com/cockroachdb/cockroach/pkg/jobs.RegisterExecutorsMetrics()
      /go/src/github.com/cockroachdb/cockroach/pkg/jobs/scheduled_job_executor.go:136 +0x1a5
  github.com/cockroachdb/cockroach/pkg/jobs.(*jobScheduler).runDaemon.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/jobs/job_scheduler.go:351 +0x176
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:442 +0x161
```

**Comment by RaduBerinde** [comment]
Saw this during TestRotateCerts and TestUseCerts.

**Comment by RaduBerinde** [comment]
@pbardea maybe related to https://github.com/cockroachdb/cockroach/pull/67855 ?

**Comment by RaduBerinde** [comment]
Maybe `Registry.MarshalJSON()` should now take the lock? Or maybe we should `RegisterExecutorsMetrics` before we start the daemon?

**Comment by RaduBerinde** [comment]
Closing in favor of duplicate https://github.com/cockroachdb/cockroach/issues/69522.
